simple_graph.py

```diff
@@ -35,10 +38,12 @@ def add_node(self, n):
         """adds a new node 'n' to the graph"""
         self.nodes_.add(n)
 
-    def add_edge(self, n1, n2):
+    def add_edge(self, n1, n2, weight):
         """adds a new edge to the graph connecting 'n1' and 'n2', if either n1
         or n2 are not already present in the graph, they should be added.
         """
+        weight = int(weight)
+
         if n1 not in self.nodes_:
             self.nodes_.add(n1)
```

Comment:
given that `self.nodes_` is a set, and sets automatically de-duplicate, is there any benefit at all to checking if `n1` is in the set of nodes before adding it?